### PR TITLE
HGDP + 1KG update metadata

### PR DESCRIPTION
**Pull request by jkgoodrich**
Function copied from this PR: https://github.com/broadinstitute/gnomad_qc/pull/390 

This contains the function for updating the v4 genomes metadata.

The first commit is basically a copy from the above PR, and any commits after that are my suggested changes.

Requires https://github.com/broadinstitute/gnomad_methods/pull/620

### CODE REVIEW

File: gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py [hunk 1]
```diff
@@ -188,6 +190,151 @@ def get_hgdp_tgp_v4_exome_duplicates(
...
+            - `gnomad_sample_filters.hard_filtered`: to apply the recomputed freemix
+              filter for HGDP samples.
+            - `gnomad_sample_filters.related_to_nonsubset`: to further filter out
+              samples that are related to samples outside the subset but were not
+              included in the v3 release.
+            - `gnomad_sample_filters.v4_exome_duplicate`: to further filter out the
+              samples in the HGDP + 1KG subset that are duplicates of an exome in the
+              v4 release.
+            - `sample_filters.relatedness_inference.related`: to apply the updated
+              relatedness inference implemented by Alicia Martin's group.
```
Comment (KoalaQin) [inline gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py:212]:
```suggestion
            - `relatedness_inference.related`: to apply the updated relatednesss
                inference implemented by Alicia Martin's group.
```

File: gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py [hunk 2]
```diff
@@ -268,6 +427,7 @@ def main(args):
         tmp_dir="gs://gnomad-tmp-4day",
     )
     v4_genome_release_resources = get_v4_genomes_release_resources(overwrite=overwrite)
+    v3_meta_ht = v4_genome_release_resources.meta_ht.ht()
```
Comment (KoalaQin) [inline gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py:430]:
```suggestion
    meta_ht = v4_genome_release_resources.meta_ht.ht()
```
I found v3_meta_ht here is confusing, I know here you're using as a global, but it is in other two args too. I prefer 'meta_ht' to be reserved for HGDP/TGP meta only.

File: gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py [hunk 3]
```diff
@@ -285,6 +445,14 @@ def main(args):
         )
         ht.write(res.hgdp_tgp_v4_exome_duplicate_ht.path, overwrite=overwrite)
 
+    if args.update_annotations:
+        res = v4_genome_release_resources.update_annotations
+        res.check_resource_existence()
+        logger.info("Adding updated sample QC annotations to meta HT...")
+        add_updated_sample_qc_annotations(v3_meta_ht).write(
```
Comment (KoalaQin) [inline gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py:452]:
```suggestion
        add_updated_sample_qc_annotations(meta_ht).write(
```

**Review by KoalaQin — CHANGES_REQUESTED** [review]
What's your thoughts on changing 'v4' to 'v4.0'? I didn't change in this PR, I think if you lean on one, just replace it.

File: gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py [hunk 4]
```diff
@@ -186,6 +192,151 @@ def get_hgdp_tgp_v4_exome_duplicates(
...
+        updated_counts["n_outlier"],
+        updated_counts["n_diff"]["hgdp_tgp_meta.subcontinental_pca.outlier"],
+        updated_counts["n_diff"]["hgdp_tgp_meta.population"],
+        updated_counts["n_related_subset"],
+        updated_counts["n_diff"]["relatedness_inference.related"],
+        updated_counts["n_related_nonsubset"],
+        updated_counts["n_duplicate_to_exomes"],
+        updated_counts["n_release"],
+    )
+    return updated_ht
```
Comment (KoalaQin) [inline gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py:337]: When I tested, this chunck of logger is not working at all. 
https://console.cloud.google.com/dataproc/jobs/9ded3f68943149a8a0de33c9cae7e547/monitoring?region=us-central1&project=broad-mpg-gnomad
Do you know why?

**Review by KoalaQin — COMMENTED** [review]
It seems 'logging' is not working.

File: gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py (hunk 4 above)
Comment (jkgoodrich) [reply gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py:337]: I think there is something wrong with the loggers all of a sudden. Mike mentioned an issue with the loggers in the release code too. I will look into it, but I don't think it needs to hold up merging

**Review by KoalaQin — APPROVED** [review]
I checked the output meta, the updates look good.

**Pull request merged**